web/src/left_sidebar_navigation_area.ts

```diff
@@ -207,30 +207,52 @@ export function highlight_all_messages_view(): void {
     }, 0);
 }
 
+// Reorder the <li> elements in the DOM based on the current home_view.
+// Called in 2 places: on initial page load and when home_view changes.
+export function reorder_left_sidebar_navigation_list(home_view: string): void {
+    const $left_sidebar = $("#left-sidebar-navigation-list");
+    const $left_sidebar_condensed = $("#left-sidebar-navigation-list-condensed");
+
+    const $inbox_rows = $(".top_left_inbox");
+    const $all_messages_rows = $(".top_left_all_messages");
+    const $recent_views_rows = $(".top_left_recent_view");
+
+    if (home_view === settings_config.web_home_view_values.all_messages.code) {
+        $all_messages_rows.detach();
+        $left_sidebar.prepend($all_messages_rows.eq(1));
+        $left_sidebar_condensed.prepend($all_messages_rows.eq(0));
+    } else if (home_view === settings_config.web_home_view_values.recent_topics.code) {
+        $recent_views_rows.detach();
+        $left_sidebar.prepend($recent_views_rows.eq(1));
+        $left_sidebar_condensed.prepend($recent_views_rows.eq(0));
+    } else {
+        $inbox_rows.detach();
+        $left_sidebar.prepend($inbox_rows.eq(1));
+        $left_sidebar_condensed.prepend($inbox_rows.eq(0));
+    }
+}
+
 function handle_home_view_order(home_view: string): void {
-    // Remove class and tabindex from current home view
+    // Remove class from current home view
     const $current_home_view = $(".selected-home-view");
     $current_home_view.removeClass("selected-home-view");
-    $current_home_view.find("a").removeAttr("tabindex");
 
+    const $inbox_rows = $(".top_left_inbox");
     const $all_messages_rows = $(".top_left_all_messages");
     const $recent_views_rows = $(".top_left_recent_view");
-    const $inbox_rows = $(".top_left_inbox");
 
     const res = unread.get_counts();
 
-    // Add the class and tabindex to the matching home view
+    // Add the class to the matching home view
     if (home_view === settings_config.web_home_view_values.all_messages.code) {
```

Comment:
I believe all of this logic around the `selected-home-view` class and its references in CSS can be removed now, as all of that was meant to shift things into view.

Please remove in a separate commit, and test thoroughly to make sure there aren't any ill side effects from that.

Reply:
No, CSS maybe yes, but not all,  [see this](https://github.com/zulip/zulip/blob/main/web/src/left_sidebar_navigation_area.ts#L70)

Reply:
Hmmm, come to think of it, it can be replacable but just updating `home_unread_messages` to the first `<li>` child